### PR TITLE
added digital matter oyster

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/digitalMatter/oyster/uplink.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  return result;
+}
+
+function decoder(bytes, port) {
+  var decoded = {};
+  if (port === 1) {
+    decoded.type = "position";
+    decoded.latitudeDeg = bytes[0] + bytes[1] * 256 +
+      bytes[2] * 65536 + bytes[3] * 16777216;
+    if (decoded.latitudeDeg >= 0x80000000) // 2^31
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.js:19]: use brackets

File: types/digitalMatter/oyster/uplink.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+// Parse Hex Byte Array
+function parseHexString(str) {
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.js:2]: use bits?

File: types/digitalMatter/oyster/uplink.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    decoded.type = "position";
+    decoded.latitudeDeg = bytes[0] + bytes[1] * 256 +
+      bytes[2] * 65536 + bytes[3] * 16777216;
+    if (decoded.latitudeDeg >= 0x80000000) // 2^31
+      decoded.latitudeDeg -= 0x100000000; // 2^32
+    decoded.latitudeDeg /= 1e7;
+
+    decoded.longitudeDeg = bytes[4] + bytes[5] * 256 +
+      bytes[6] * 65536 + bytes[7] * 16777216;
+    if (decoded.longitudeDeg >= 0x80000000) // 2^31
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.js:25]: brackets

File: types/digitalMatter/oyster/uplink.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  }
+  else if (port === 4) {
+    decoded.type = "position";
+    decoded.latitudeDeg = bytes[0] + bytes[1] * 256 + bytes[2] * 65536;
+    if (decoded.latitudeDeg >= 0x800000) // 2^23
+      decoded.latitudeDeg -= 0x1000000; // 2^24
+    decoded.latitudeDeg *= 256e-7;
+
+    decoded.longitudeDeg = bytes[3] + bytes[4] * 256 + bytes[5] * 65536;
+    if (decoded.longitudeDeg >= 0x800000) // 2^23
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.js:43]: brackets

File: types/digitalMatter/oyster/lifecycle.schema.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      "description": "Seconds spent per successful GPS fix, not including freshen time"
+    },
+    "aveGpsFailS": {
+      "title": "Average GPS fail time",
+      "type": "integer",
+      "unit": "s",
+      "description": "Seconds spent per failed GPS fix attempt"
+    },
+    "aveGpsFreshenS": {
+      "title": "Average GPS freshening time",
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/lifecycle.schema.json:48]: freshening?

File: types/digitalMatter/oyster/lifecycle.schema.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      "minimum": 0,
+      "maximum": 24,
+      "title": "Initial battery voltage",
+      "desciption": "Initial battery voltage",
+      "unit": "V"
+    },
+    "txCount": {
+      "title": "Tx count",
+      "type": "integer",
+      "description": "Transmition count"
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/lifecycle.schema.json:18]: typo transmission

File: types/digitalMatter/oyster/downlink_ack.schema.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "$id": "https://akenza.io/digitalMatter/oyster/position.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Position",
+  "type": "object",
+  "properties": {
+    "sequence": {
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/downlink_ack.schema.json:7]: rename to sequenceNumber

File: types/digitalMatter/oyster/position.schema.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    "speedKmph": {
+      "title": "speedKmph",
+      "unit": "kmh",
+      "type": "number",
+      "description": "Acceleration of the sensor"
+    },
+    "manDown": {
+      "title": "Man down",
+      "type": "boolean",
+      "description": "equals true if no movement for configured period got detected (Needs to be activated)"
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/position.schema.json:48]: always start with capitals

**Review by mariusgiger — CHANGES_REQUESTED** [review]
some comments

**Comment by J2thatsme** [comment]
@mariusgiger please check again

File: types/digitalMatter/oyster/uplink.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    decoded.aveGpsFreshenS = 1 * ((bytes[7] >> 8) + (bytes[8] & 0xFF) * 1);
+    decoded.wakeupsPerTrip = 1 * ((bytes[8] >> 8) + (bytes[9] & 0x7F) * 1);
+    decoded.uptimeWeeks = 1 * ((bytes[9] >> 7) + (bytes[10] & 0xFF) * 2);
+  }
+
+  return decoded;
+}
+
+function consume(event) {
+  var payload = event.data.payload_hex;
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.js:71]: isn't it always payloadHex?

File: types/digitalMatter/oyster/uplink.spec.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  );
+});
+
+describe("Digital matter Oyster Uplink", function () {
+  describe("consume()", function () {
+    it("should decode the digital matter Oyster payload", function () {
+      const data = {
+        data: {
+          port: 1,
```
Comment (mariusgiger) [inline types/digitalMatter/oyster/uplink.spec.js:36]: please also add tests for the other ports

**Review by mariusgiger — DISMISSED** [review]
looks clean, some comments

**Comment by mariusgiger** [comment]
@J2thatsme please add a changelog entry

File: types/digitalMatter/oyster/uplink.js (hunk 9 above)
Comment (J2thatsme) [reply types/digitalMatter/oyster/uplink.js:71]: No

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**